app/controllers/api/v2/app_group_users_controller.rb

```diff
@@ -0,0 +1,34 @@
+class Api::V2::AppGroupUsersController < Api::V2::BaseController
+  include Wisper::Publisher
+
+  before_action :validate_params
+  before_action :set_app_group
+  before_action :set_user
+
+  def create
+    app_group_role = AppGroupRole.find_by_name(params[:app_group_role])
+    render(json: { success: false, errors: ['App group role not found'], data: nil }, status: :not_found) && return if app_group_role.blank?
+
+    app_group_user = AppGroupUser.create(app_group_id: @app_group.id, user_id: @user.id, role_id: app_group_role.id)
+    render(json: { success: false, errors: [app_group_user.errors], data: nil }, status: :unprocessable_entity) && return unless app_group_user.valid?
+
+    render json: { success: true, errors: nil, data: ['App group user created'] }, status: :ok
+  end
+
+  private
+
+  def validate_params
+    valid, error_response = validate_required_keys([:app_group_secret, :app_group_role, :user_email])
+    render json: error_response, status: error_response[:code] unless valid
+  end
+
+  def set_app_group
+    @app_group = AppGroup.find_by(secret_key: params[:app_group_secret])
+    render json: { success: false, errors: ['App Group not found'], data: nil }, status: :not_found if @app_group.blank?
+  end
+
+  def set_user
+    @user = User.find_by_username_or_email(params[:user_email])
```

Comment:
Currently, Barito doesn't save email from gate user, we only save the username. If I merge this PR and you use this with email, you will always get error because the email is empty.

Reply:
So, if I pass the `user_name` as param in the API and use the same for finding the user. It will be able to find the gate user right?

Reply:
Yes Mas.

Reply:
Updated the API to use gate_username instead of user email